Modules/cjkcodecs/multibytecodec.c

```diff
@@ -895,6 +897,96 @@ _multibytecodec_MultibyteIncrementalEncoder_encode_impl(MultibyteIncrementalEnco
     return encoder_encode_stateful(STATEFUL_ECTX(self), input, final);
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_getstate_impl(MultibyteIncrementalEncoderObject *self)
+/*[clinic end generated code: output=9794a5ace70d7048 input=4a2a82874ffa40bb]*/
+{
+    const char *pendingbuffer;
+    Py_ssize_t pendingsize;
+
+    if (self->pending != NULL) {
+        pendingbuffer = PyUnicode_AsUTF8AndSize(self->pending, &pendingsize);
+        if (pendingbuffer == NULL) {
+            return NULL;
+        }
+
+        return _PyLong_FromByteArray((const unsigned char *)pendingbuffer,
+                                     pendingsize, PY_LITTLE_ENDIAN, 0);
+    } else {
+        return _PyLong_Zero;
+    }
+}
+
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.setstate
+    state: object
+    /
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_setstate(MultibyteIncrementalEncoderObject *self,
+                                                     PyObject *state)
+/*[clinic end generated code: output=2ef45090fc274f1e input=c9e279e65a2e1c8d]*/
+{
+    PyObject *pending;
+    PyObject *ret;
+    PyLongObject *statelong;
+    unsigned char pendingbuffer[MAXENCPENDING];
+
+    if (PyLong_AsLong(state) != 0) {
+        /* convert long into pending bytes */
+
+        statelong = (PyLongObject *)PyNumber_Long(state);
+        if (statelong == NULL) {
+            return NULL;
+        }
+
+        if (_PyLong_AsByteArray(statelong, pendingbuffer,
+                                sizeof(pendingbuffer),
+                                PY_LITTLE_ENDIAN, 0) < 0) {
+            Py_DECREF(statelong);
```

Comment:
Is it possible to refactor all error paths under one "error" label section and substitute these with gotosas indicated [here](https://docs.python.org/3/c-api/intro.html) so there is only one exit point from the function?